### PR TITLE
Add proxy support

### LINKED ISSUE #171: Add method to set proxy
Similar to Java SDK: https://github.com/keenlabs/KeenClient-Java#using-an-http-proxy

### PR DESCRIPTION
**Pull request by heitortsergent**
This PR closes #171.

This change will require users to add `CFNetwork.framework` to their binaries... :/ I'm trying to find a way to make this optional.

### CODE REVIEW

File: KeenClient/KeenClient.m [hunk 1]
```diff
@@ -239,27 +231,7 @@ + (KIODBStore *) getDBStore {
...
-    self.queryQueue = dispatch_queue_create(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
-
-    self.maxEventUploadAttempts = 3;
-    
-    self.maxQueryAttempts = 10;
-    
-    self.queryTTL = 3600;
-
-    return self;
+    return [self initWithProjectID:@"pi" andWriteKey:@"wk" andReadKey:@"rk"];
```
Comment (terryhorner) [inline KeenClient/KeenClient.m]: I'm not sure we want to initialize with default values like this, it could lead to confusion. Might be better to initialize with empty values and verify in `initWithProjectID` that they're not empty.

File: KeenClient/KeenClient.m [hunk 2]
```diff
@@ -300,7 +287,7 @@ - (id)initWithProjectID:(NSString *)projectID andWriteKey:(NSString *)writeKey a
             self.readKey = readKey;
         }
     }
-
+
```
Comment (terryhorner) [inline KeenClient/KeenClient.m]: Nit: unnecessary whitespace.

**Comment by heitortsergent** [comment]
@terryhorner thanks for the comments! Made the changes, I went back to the previous `init` implementation, just added a `if (self)` check to it.

**Comment by terryhorner** [comment]
👍

**Comment by heitortsergent** [comment]
@terryhorner finding a way to make it optional is taking longer than I expected. :/

In the meantime, do you think we could merge this PR and also add a 'beta' release tag to it, so users could test it out? (and can you approve the review please? :shipit:)

**Comment by terryhorner** [comment]
I am hesitant to approve a beta release due to the fact that it could find its way into apps that get pushed to the App Store only to never be updated in the future.

Is it possible to make `CFNetwork.framework` optional? If not, we'll have to deal with it as is.

**Comment by heitortsergent** [comment]
@terryhorner that's a good point...

The only working possibility I've found so far requires the users to add CFNetwork to their project build settings, and there they can mark it as optional. But that will break the build of everyone who updates the library, and the error Xcode throws is not really clear.

I'm trying to find if there are any alternatives methods that don't break existing builds.

**Comment by baumatron** [comment]
#227 replaces this
